src/plugins/data/public/mocks.ts

```diff
@@ -16,34 +16,30 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { Plugin } from '.';
 import { searchSetupMock } from './search/mocks';
 import { queryServiceMock } from './query/mocks';
 
-export type Setup = jest.Mocked<ReturnType<Plugin['setup']>>;
```

Comment:
I'm curious, why do you remove these interfaces? Idea for these mocks is to have same interface/"shape" as the real plugin.

Now anyone can add anything to those mocks:

![image](https://user-images.githubusercontent.com/9773803/67938017-fa23dd80-fbce-11e9-82b9-e7aab6657a51.png)

Whereas before TypeScript would find an error:

![image](https://user-images.githubusercontent.com/9773803/67938101-22134100-fbcf-11e9-9e98-e66a1edc21db.png)

Reply:
It's actually causing a very annoying bug when using it.
Can explain in person

Reply:
Managed to revert this